### PR TITLE
Add definition for swagger-jsdoc

### PR DESCRIPTION
**Pull request by drGrove**
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

**Comment by vvakame** [comment]
please rename swagger-jsdoc/swagger-jsdoc-test.ts to swagger-jsdoc/swagger-jsdoc-tests.ts

**Comment by drGrove** [comment]
File name changed and revisions made to fix breaking tests

File: swagger-jsdoc/swagger-jsdoc-tests.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/// <reference path="swagger-jsdoc.d.ts" />
+/// <reference path="../express/express.d.ts" />
+
+import * as express from 'express';
+const swaggerJsdoc = require('./swagger-jsdoc');
```
Comment (andy-ms) [inline swagger-jsdoc/swagger-jsdoc-tests.ts]: `require` is a function returning `any`. Please use `import swaggerJsdoc = require("./swagger-jsdoc");` instead.

File: swagger-jsdoc/swagger-jsdoc.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            title: string,
+            version: string
+        },
+        host: string,
+        basePath: string
+    },
+    apis: string[]
+}
+
+declare interface swaggerjsdoc {
```
Comment (andy-ms) [inline swagger-jsdoc/swagger-jsdoc.d.ts]: Use a module declaration here:
```ts
declare module "swagger-jsdoc" {
  function swaggerJsDoc(options?: swaggerOptions): any;
  export = swaggerJsDoc;
}
```

**Comment by drGrove** [comment]
@andy-ms when I make the requested changes, I get swagger-jsdoc.d.ts is not a module error. 

Any suggestions?

**Comment by drGrove** [comment]
@andy-ms 
When I make the requested changes, i'm getting 'swagger-jsdoc.d.ts' is not a module error.

I'm new to typescript, so any help would be great.

**Comment by vvakame** [comment]
@drGrove can you show me the repro code?

**Comment by andy-ms** [comment]
@drGrove This reminds me of #10857. You should be able to fix this using `--module commonjs`.

**Comment by drGrove** [comment]
@vvakame, he's the updated code. 

@andy-ms, attempted --module commonjs and still getting 'not a module'

**Comment by andy-ms** [comment]
Use `require('swagger-jsdoc')` instead of `require('./swagger-jsdoc')`.

The compiler is giving you the error because "./swagger-jsdoc" is a `.d.ts` file that defines a global module inside of it; it's not itself a module. So you use the `/// <reference path="./swagger-jsdoc.d.ts" />` to pick up the local file, but `require('swagger-jsdoc')` to use the globally defined module obtained from there.

(In the `types-2.0` branch this is simpler and we will have a `index.d.ts` which directly defines exports rather than having a module inside of it. We'll do this change for you when merging changes from `master` to `types-2.0`.)

Also, you'll need a file `swagger-jsdoc-tests.ts.tscparams` with the content `-m commonjs` to get the test runner to use the correct options.

**Comment by drGrove** [comment]
@andy-ms @vvakame  this should be ready to review now.

File: swagger-jsdoc/swagger-jsdoc.d.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Type definitions for Swagger-JSDoc
+// Project: https://github.com/surnet/swagger-jsdoc
+// Definitions by: Daniel Grove <https://github.com/drGrove>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/* =================== USAGE ===================
+
+    import * as Express from "express"
+    import swaggerJsdoc = require('./swagger-jsdoc');
```
Comment (vvakame) [inline swagger-jsdoc/swagger-jsdoc.d.ts]: please update this line.

**Comment by drGrove** [comment]
@vvakame, Latest changes up.

**Pull request merged**